pootle/apps/pootle_store/views.py

```diff
@@ -413,6 +411,7 @@ def get_units(request):
                 begin = max(index - chunk_size, 0)
                 end = min(index + chunk_size + 1, len(uid_list))
                 uids = uid_list[begin:end]
+                units = step_queryset[begin:end]
```

Comment:
no - calling filter a few lines below is *wrong* and is partly to blame for random results

when you filter pk__in=somelist - you will not get the results back in the order of somelist from a real db (altough you do from the in-mem test db)